addons/sys_components/fnc_getAntennaDirMan.sqf

```diff
@@ -13,23 +13,55 @@
  *
  * Public: No
  */
+#include "script_component.hpp"
 
- private ["_forwardV", "_upV"];
- params ["_obj"];
+private ["_forwardV", "_upV"];
+params ["_obj"];
 
- //@TODO: This is a hack fix for vehicles having funky up vectors when people are inside...
- if (vehicle _obj == _obj) then {
-     private _spinePos = _obj modelToWorldVisual (_obj selectionPosition "Spine3");
-     _upV = _spinePos vectorFromTo (_obj modelToWorldVisual (_obj selectionPosition "Neck"));
+//@TODO: This is a hack fix for vehicles having funky up vectors when people are inside...
+if (vehicle _obj == _obj) then {
+    private _spinePos = _obj modelToWorldVisual (_obj selectionPosition "Spine3");
+    
+    drawIcon3D ["a3\ui_f\data\gui\Rsc\RscDisplayArsenal\radio_ca.paa", [1,1,1,1], _spinePos, 1, 1, 45, "Spine", 1, 0.05, "TahomaB"];
 
-     private _upP = _upV call cba_fnc_vect2polar;
-     _upP set [2, (_upP select 2) - 90];
-     _forwardV = _upP call cba_fnc_polar2vect;
+    _upV = _spinePos vectorFromTo (_obj modelToWorldVisual (_obj selectionPosition "Neck"));
 
-     _forwardV = (ATLtoASL _spinePos) vectorFromTo (ATLtoASL (_spinePos vectorAdd _forwardV));
-     _upV = (ATLtoASL _spinePos) vectorFromTo (ATLtoASL (_spinePos vectorAdd _upV));
- } else {
-     _forwardV = vectorDir (vehicle _obj);
-     _upV = vectorUp (vehicle _obj);
- };
- [_forwardV, _upV];
+    private _upP = _upV call cba_fnc_vect2polar;
+    //hint format ["%1\n%2", _upV, _upP];
+    
+    if (EGVAR(sys_core,automaticAntennaDirection)) then {
```

Comment:
Can we make it a function in order to hook it with deployable antennas? I will modify the function directly to account for it. It could have the same name.

Reply:
Yes. From here I can modify it, if necessary, for external radios (dead unit) or deployable antennas.